### PR TITLE
Successfully implemented sorting feature, put and post request, and review tile features

### PR DESCRIPTION
**Pull request by moneycabui**

### CODE REVIEW

File: client/src/components/ReviewsList/index.jsx [hunk 1]
```diff
@@ -31,24 +35,67 @@ class ReviewsList extends React.Component {
...
     const { currentProduct } = this.props;
+    const { sortList } = this.state;
     const { currentProduct: previousProduct } = prev;
+    const { sortList:  priorSortList } = previousState;
     if (currentProduct !== previousProduct) {
       this.getReviews();
     }
+    if (sortList !== priorSortList) {
+      this.getReviews();
+    }
```
Comment (phil-ho) [inline client/src/components/ReviewsList/index.jsx:48]: Next time it's better make sure you only call the API once by declaring a variable like `shouldGetReviews` and conditionally set it to true depending on the situation. Then at the end, check if you should call the API and call it.

```suggestion
    let shouldGetReviews = false;
    
    if (currentProduct !== previousProduct) {
      shouldGetReviews = true;
    }
    if (sortList !== priorSortList) {
      shouldGetReviews = true;
    }
    
    if (shouldGetReviews === true) {
      this.getReviews();
    }
```

File: client/src/components/ReviewsList/index.jsx [hunk 2]
```diff
@@ -220,53 +263,89 @@ class ReviewsList extends React.Component {
...
-              Total Reviews
+              <div className={styles.recommendOverview}>
+                <div className={styles.recommendedPercent}>
+                  {recommendPercent}
+                  %
+                </div>
+                {' '}
+                of reviewers recommend this product
+              </div>
+              <ReviewRatingDistribution
```
Comment (phil-ho) [inline client/src/components/ReviewsList/index.jsx:312]: the same util we used to move starRatings out of App state can be applied here if you have time

**Review by phil-ho — APPROVED** [review]
Your hard work is definitely paying off, the component is looking very good

File: client/src/components/ReviewsList/index.jsx (hunk 2 above)
Comment (moneycabui) [reply client/src/components/ReviewsList/index.jsx:312]: Ooo that's a good idea actually - will attempt in the morning! Definitely appreciate the feedback 😊

**Review by agentanky — APPROVED** [review]

**Pull request merged**